pkg/smartcontract/parameter.go

```diff
@@ -63,8 +63,16 @@ func (p *Parameter) MarshalJSON() ([]byte, error) {
 		resultErr      error
 	)
 	switch p.Type {
-	case BoolType, IntegerType, StringType, Hash256Type, Hash160Type:
+	case BoolType, StringType, Hash256Type, Hash160Type:
 		resultRawValue, resultErr = json.Marshal(p.Value)
+	case IntegerType:
+		val, ok := p.Value.(int64)
```

Comment:
I'm not sure it's correct to always expect an int64 here. Look at our own `adjustValToType` for example.

Reply:
I'd rather fix `adjustValToType`, as `.Value` is our internal representation and it is better be unique. All conversions could be performed during unmarshaling.

Reply:
OK.